### PR TITLE
Fix malloc random size fragment test

### PR DESCRIPTION
**Pull request by larrywxyz**
#2776 

Seems like this issue got automatically closed when the PR closed, but it was never actually fixed. 

Basically, I think the problem with the test is that when random size memory being malloced is too large (in this case within 23 bytes of _get_heap_size()), dlmalloc will automatically increase the size of the heap. After the heap reaches its maximum size (2097024 bytes), the heap is shrunk back down (3968 bytes), before once again increasing in size if _randx(_get_heap_size()) is too large.

The simple fix is to restrict the max possible size being allocated in the random size fragment test.

### CODE REVIEW

File: tests/memory/enc/fragment.c [hunk 1]
```diff
@@ -80,7 +80,7 @@ static int _malloc_free_random_size(int times, unsigned int seed)
     srand(seed);
     for (i = 0; i < times; i++)
     {
-        size = _randx(_get_heap_size());
+        size = _randx(_get_heap_size() - 23);
```
Comment (mingweishih) [inline tests/memory/enc/fragment.c:83]: Thanks for investigate this! It seems that the problem is that the test hits a corner case (i.e., allocating certain large size heap). Rather than avoiding hitting the case, do you think we can fundamentally address the problem? I'd suggest as follows.
- Investigate if there is a fix (maybe certain configuration to the dlmalloc).
- Add tests to explicitly test this corner case. In this case, we should know what would be the expected behavior.

**Comment by anakrish** [comment]
@larrywxyz 
>  dlmalloc will automatically increase the size of the heap

What does this mean within an enclave? The size of the enclave heap is a fixed-value.

File: tests/memory/enc/fragment.c (hunk 1 above)
Comment (radhikaj) [reply tests/memory/enc/fragment.c:83]: @larrywxyz  Can you address this comment?

**Comment by radhikaj** [comment]
Thanks @larrywxyz Please reopen when you are able to work on this again.
